vendor/engines/bulk_email/app/decorators/bulk_email/job_decorator.rb

```diff
@@ -0,0 +1,42 @@
+module BulkEmail
+
+  class JobDecorator < SimpleDelegator
+
+    def sender
+      user.email
+    end
+
+    def recipient_list
+      recipients.join(", ")
+    end
+
+    def products
+      Product.find(selected_product_ids).map(&:name).join(", ")
```

Comment:
Seems like it might have more to do with a blank `product_id`. If I have a product in the search_criteria, but a blank `product_id`, then I'm getting the same error.